compendium/DeclarativeServices/test/gtest/TestAsyncWorkService.cpp

```diff
@@ -367,4 +366,189 @@ namespace test
         ASSERT_TRUE(service) << "GetService failed for CAInterface.";
     }
 
+    class Barrier
+    {
+      public:
+        Barrier(std::size_t count) : threshold(count), count(count), generation(0) {}
```

Comment:
this is duplicate code that already exists in compendium\DeclarativeServices\test\gtest\TestComponentConfigurationImpl.cpp. Please place this class in a general location to be used by all tests.